pkg/storagepolicy/sdkstoragepolicy.go

```diff
@@ -0,0 +1,263 @@
+
+	errPolicyManagerNotInitialized = errors.New("Policy Manager is not initialized")
+	errPolicyManagerInitialized    = errors.New("Policy Manager is already initialized")
+
+	inst *SdkPolicyManager
+	Inst = func() (PolicyManager, error) {
+		return policyInst()
+	}
+)
+
+func Init(kv kvdb.Kvdb) error {
+	if inst != nil {
+		return errPolicyManagerInitialized
+	}
+	if kv == nil {
+		return errors.New("KVDB is not yet initialized.  " +
+			"A valid KVDB instance required for the Storage Policy.")
+	}
+
+	inst = &SdkPolicyManager{
+		kv: kv,
+	}
+
+	return nil
+}
+
+func policyInst() (PolicyManager, error) {
+	if inst == nil {
+		return nil, errPolicyManagerNotInitialized
+	}
+	return inst, nil
+}
+
+// Simple function which creates key for Kvdb
+func prefixWithName(name string) string {
+	return policyPrefix + policyPath + "/" + name
+}
+
+func (p *SdkPolicyManager) Create(
+	ctx context.Context,
+	req *api.SdkOpenStoragePolicyCreateRequest,
+) (*api.SdkOpenStoragePolicyCreateResponse, error) {
+	if req.StoragePolicy.GetName() == "" {
+		return nil, status.Error(codes.InvalidArgument, "Must supply a Storage Policy Name")
+	}
+
+	if req.StoragePolicy.GetPolicy() == nil {
+		return nil, status.Error(codes.InvalidArgument, "Must supply Volume Specs")
+	}
+
+	// Since VolumeSpecUpdate has oneof method of proto,
+	// we need to marshal it into string using protobuf jsonpb
+	m := jsonpb.Marshaler{}
+	policyStr, err := m.MarshalToString(req.StoragePolicy.GetPolicy())
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Json Marshal failed for policy %s: %v", req.StoragePolicy.GetName(), err)
+	}
+
+	_, err = p.kv.Create(prefixWithName(req.StoragePolicy.GetName()), policyStr, 0)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Failed to save storage policy: %v", err)
+	}
+
+	return &api.SdkOpenStoragePolicyCreateResponse{}, nil
+}
+
+func (p *SdkPolicyManager) Update(
+	ctx context.Context,
+	req *api.SdkOpenStoragePolicyUpdateRequest,
+) (*api.SdkOpenStoragePolicyUpdateResponse, error) {
+	if req.StoragePolicy.GetName() == "" {
+		return nil, status.Error(codes.InvalidArgument, "Must supply a Storage Policy Name")
+	}
+
+	if req.StoragePolicy.GetPolicy() == nil {
+		return nil, status.Error(codes.InvalidArgument, "Must supply Volume Specs")
+	}
+
+	m := jsonpb.Marshaler{}
+	policyStr, err := m.MarshalToString(req.StoragePolicy.GetPolicy())
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Json Marshal failed for policy %s: %v", req.StoragePolicy.GetName(), err)
+	}
+
+	_, err = p.kv.Update(prefixWithName(req.StoragePolicy.GetName()), policyStr, 0)
+	if err == kvdb.ErrNotFound {
+		return nil, status.Errorf(codes.NotFound, "Storage Policy %s not found", req.StoragePolicy.GetPolicy())
+	} else if err != nil {
+		return nil, status.Errorf(codes.Internal, "Failed to update storage policy: %v", err)
+	}
+
+	return &api.SdkOpenStoragePolicyUpdateResponse{}, nil
+}
+
+func (p *SdkPolicyManager) Delete(
```

Comment:
If the name being deleted is also enforced, then this should call the Release() function.